### DISCUSSION TITLE
How to update camera up vector on button press

### DISCUSSION THREAD

**Discussion by radcli14** [google/filament#5009] [Q&A]
I am working on an Android app that uses a variation on the ModelViewer class to render several .glb formatted models. I would like to allow the user to change the default camera orientation by tapping a button in the corner of the screen. I am attempting to implement this as follows:

```
// Update the up vector when the user taps the button in the lower right
view.findViewById<Button>(R.id.view_button).setOnClickListener {
    when {
        viewModel.upward[0] == 1.0 -> viewModel.upward = arrayOf(0.0, 1.0, 0.0)
        viewModel.upward[1] == 1.0 -> viewModel.upward = arrayOf(0.0, 0.0, 1.0)
        viewModel.upward[2] == 1.0 -> viewModel.upward = arrayOf(1.0, 0.0, 0.0)
    }
    val targetPosition = viewModel.defaultTargetPosition()
    val orbitHomePosition = viewModel.defaultOrbitHomePosition()
    modelViewer.camera.lookAt(
        orbitHomePosition[0].toDouble(), orbitHomePosition[1].toDouble(), orbitHomePosition[2].toDouble(),
        targetPosition[0].toDouble(), targetPosition[1].toDouble(), targetPosition[2].toDouble(),
        viewModel.upward[0], viewModel.upward[1], viewModel.upward[2])
}
```

What I'm finding is that this apparently does nothing, at least not immediately, but if I refresh the fragment by navigating to a different tab and back, then my up vector will be set to the intended value. Any ideas on the correct way to do this?

**Comment by swafe** [comment]
Have you found a solution?
